ci-operator/step-registry/cucushift/upgrade/arch-migration/cucushift-upgrade-arch-migration-commands.sh

```diff
@@ -0,0 +1,330 @@
+
+    #2. Check all cluster operators get stable and ready
+    echo "Step #2: check all cluster operators get stable and ready"
+    wait_clusteroperators_continous_success
+
+    #3. Make sure every machine is in 'Ready' status
+    echo "Step #3: Make sure every machine is in 'Ready' status"
+    check_node
+
+    #4. All pods are in status running or complete
+    echo "Step #4: check all pods are in status running or complete"
+    check_pod
+}
+
+# Monitor the migration status
+function check_migrate_status() {
+    local wait_migrate="${TIMEOUT}" out avail progress
+    while (( wait_migrate > 0 )); do
+        sleep 5m
+        (( wait_migrate -= 5 ))
+        if ! ( echo "oc get clusterversion" && oc get clusterversion ); then
+            continue
+        fi
+        if ! out="$(oc get clusterversion --no-headers)"; then continue; fi
+        avail="$(echo "${out}" | awk '{print $3}')"
+        progress="$(echo "${out}" | awk '{print $4}')"
+        if [[ ${avail} == "True" && ${progress} == "False" && ${out} == *"Cluster version is ${SOURCE_VERSION}" ]]; then
+            echo -e "Migrate succeed\n\n"
+            return 0
+        fi
+    done
+    if (( wait_migrate <= 0 )); then
+        echo >&2 "Migrate timeout, exiting" && return 1
+    fi
+}
+
+# Check ClusterVersion has architecture="Multi"
+function check_arch() {
+    local msg
+    msg=$(oc get clusterversion version -ojson|jq -r '.status.conditions[]|select(.type == "ReleaseAccepted")|.message')
+    if [[ ${msg} == *"architecture=\"Multi\""* ]]; then
+        echo "ClusterVersion architecture check PASSED"
+    else
+        echo >&2 "ClusterVersion architecture check FAILED, exiting" && return 1
+    fi
+}
+
+function switch_channel() {
+    echo "Switch upgrade channel to candidate-${SOURCE_XY_VERSION}..."
+    oc adm upgrade channel candidate-${SOURCE_XY_VERSION}
+    ret=$(oc get clusterversion/version -ojson | jq -r '.spec.channel')
+    if [[ "${ret}" != "candidate-${SOURCE_XY_VERSION}" ]]; then
+        echo >&2 "Failed to switch channel, exiting" && return 1
+    fi
+}
+
+function migrate() {
+    echo "Arch migrating"
+    cmd="oc adm upgrade --to-multi-arch"
+    echo "Migrating cluster: ${cmd}"
+    if ! eval "${cmd}"; then
+        echo >&2 "Failed to request arch migration, exiting" && return 1
+    fi
+}
+
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+export OC="run_command_oc"
+
+# Target version oc will be extract in the /tmp/client directory, use it first
+mkdir -p /tmp/client /tmp/jq
+export OC_DIR="/tmp/client"
+export JQ_DIR="/tmp/jq"
+export PATH=${OC_DIR}:${JQ_DIR}:$PATH
+
+curl -L https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 -o ${JQ_DIR}/jq && chmod +x ${JQ_DIR}/jq
```

Comment:
A bit concern, if this download get broken when multiple concurrent running jobs because of github network limit. It is possible to add the jq image to `verification-tests` image?

Reply:
If this path moves forward (I imagine this is done elsewhere in the cucushift steps as well), lets consider to use the latest versions of yq... There is a (slow due to complexity) ongoing effort to remove from this repository any similar statements that download different versions of yq/jq. Since yq works on yaml and YAML is a superset of JSON, i'd reduce to use one of the latest versions of yq rather than older versions of any of these tools for new code

Reply:
@jianlinliu We've had a ticket about it. https://issues.redhat.com/browse/OCPQE-11640. @liangxia do you have plan on it or prefer me to do it?

Reply:
There is no plan to add ansible to the image at the moment, but for jq/yq, it's tracked in https://issues.redhat.com/browse/OCPQE-14089

Reply:
Thank you Liang! I'll get upgrade step updated after above ticket is done.